src/orquestra/quantum/circuits/_gates.py

```diff
@@ -480,6 +482,14 @@ def bind(self, symbols_map: Dict[sympy.Symbol, Parameter]) -> "Gate":
     def replace_params(self, new_params: Tuple[Parameter, ...]) -> "Gate":
         return self.wrapped_gate.replace_params(new_params).power(self.exponent)
 
+    def __str__(self):
+        if test_has_non_commuting_gate_type(self, [Exponential]):
```

Comment:
[suggestion] Perhaps rename `test` to `check` – as `test` has connotations with, well... testing? I always get a momentary confusion when I see the word `test` in non-test code :)

Reply:
👍